src/enigma2/ConnectionManager.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+ *      Copyright (C) 2005-2011 Team XBMC
+ *      http://www.xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, write to
+ *  the Free Software Foundation, 675 Mass Ave, Cambridge, MA 02139, USA.
+ *  http://www.gnu.org/copyleft/gpl.html
+ *
+ */
+
+#include "ConnectionManager.h"
+
+// extern "C"
+// {
+// #include "libhts/htsmsg_binary.h"
+// #include "libhts/sha1.h"
+// }
+
+#include "p8-platform/os.h"
+#include "p8-platform/util/StringUtils.h"
+
+#include "../client.h"
+#include "IConnectionListener.h"
+#include "Settings.h"
+#include "utilities/Logger.h"
+#include "utilities/WebUtils.h"
+
+using namespace P8PLATFORM;
+using namespace enigma2;
+using namespace enigma2::utilities;
+
+#define FAST_RECONNECT_ATTEMPTS     (5)
+#define FAST_RECONNECT_INTERVAL   (500) // ms
+#define SLOW_RECONNECT_INTERVAL  (5000) // ms
+
+/*
+ * HTSP Connection handler
+ */
```

Comment:
HTSP? Nope? This is tvheadend comm protocol...

Reply:
Done